Documentation: Add 'Known Bugs' (#285)
OpenTabletDriver/opentabletdriver.github.io: site/_wiki/Documentation/KnownBugs.md

---
title: Known Bugs
---

## Introduction {#introduction}

OpenTabletDriver still has some fundamental and sometimes hard to fix bugs.
This page tries to document the bigger ones.

The release version of OpenTabletDriver is based on the 0.6.x branch.
The bugs described here are only relevant for the release version.

Many of these bugs have a planned milestone to be fixed in.

## Notable OpenTabletDriver bugs {#notable-bugs}

These bugs are likely to affect a sizable portion of users

### Monitor layout changes are not automatically detected {#monitor-layout-changes-not-detected}

If you change your monitor layout (such as resolution, position, rotation, or
similar) in your system settings, you need to restart the OpenTabletDriver
daemon completely.

See [OpenTabletDriver issue #1143](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/1143)
for more details.

> ### Bindings do not accomodate selected output mode (Mouse buttons not working in Windows Ink or Linux Artist Mode) {#eto-keyboard-reading-is-weird}

Reviewer: ```suggestion
### Bindings do not accommodate selected output mode (Mouse buttons not working in Windows Ink or Linux Artist Mode) {#eto-keyboard-reading-is-weird}
```

If it turns out that this doesn't work anyway, make sure the virtual keyboard
from OpenTabletDriver is assigned the US-American layout.

##### Sway window manager {#multikey-binding-keyboard-layout-linux-sway}

Put this *after* any other keyboard configuration:

```conf
input '0:0:OpenTabletDriver_Virtual_Keyboard' {
    xkb_layout us
}
```

### Some tablets may not work coming out of system sleep/suspend {#sleep-wakeup-is-inconsistent}

Replugging the device always fixes this.

See [OpenTabletDriver issue #3171](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/3171)
for more details.

### Disconnecting tablets mid-event (e.g. button press) causes the event to remain held (button permanently pressed) {#disconnected-tablets-do-not-reset-pipeline}

The only known workaround is to restart the daemon or send the same type of
event again (e.g. from the reconnected or another device)

See [OpenTabletDriver issue #2957](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/2957)
for more details.

## Assorted OpenTabletDriver Bugs {#general-bugs}

These bugs are unlikely to be encountered by users, but have been seen in the wild.

> ### Similar tablets are not separately addressible {#single-tablet-per-tablet-model}

Reviewer: ```suggestion
### Similar tablets are not separately addressable {#single-tablet-per-tablet-model}
```

---
title: Known Bugs
---

## Introduction {#introduction}

OpenTabletDriver still has some fundamental and sometimes hard to fix bugs.
This page tries to document the bigger ones.

The `0.6.x` branch is based on the GUI framework Eto, and the `avalonia`

> branch, which will eventually be migrated into `master`, is based on the GUI

Reviewer: ```suggestion
branch, which will eventually be migrated into `master`, it is based on the GUI
```

Author: The sentence reads a bit weird but I don't think it makes sense with this change. The full sentence currently reads as such:

> The `0.6.x` branch is based on the GUI framework Eto, and the `avalonia` branch, which will eventually be migrated into `master`, is based on the GUI framework Avalonia.

so the `master` tidbit is supposed to read as an inserted sentence, which I guess doesn't read great.

How about:

> The release version of OpenTabletDriver is based on the `0.6.x` branch. The bugs described here are only relevant for the release version.

I don't think it's necessarily relevant to talk about the other branches here anyway.

Author: Reworded as above in 65d1fde

framework Avalonia.

Many of these bugs have a planned milestone to be fixed in.

## Notable OpenTabletDriver bugs {#notable-bugs}

These bugs are likely to affect a sizable portion of users

### Monitor layout changes are not automatically detected {#monitor-layout-changes-not-detected}

If you change your monitor layout (such as resolution, position, rotation, or
similar) in your system settings, you need to restart the OpenTabletDriver
daemon completely.

See [OpenTabletDriver issue #1143](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/1143)
for more details.

> ### Bindings do not accomodate select output mode (Mouse buttons not working in Windows Ink or Linux Artist Mode) {#eto-keyboard-reading-is-weird}

Reviewer: Either:

```suggestion
### Bindings do not accomodate the selected output mode (Mouse buttons not working in Windows Ink or Linux Artist Mode) {#eto-keyboard-reading-is-weird}
```

```suggestion
### Bindings do not accomodate selected output mode (Mouse buttons not working in Windows Ink or Linux Artist Mode) {#eto-keyboard-reading-is-weird}
```

Author: Oopsie daisy

While Adaptive Bindings try to solve this issue, users using specific bindings
(e.g. Mouse Button Binding on Windows Ink/Linux Artist Mode) will find that
their bindings are not working.

To help users diagnose this, OpenTabletDriver v0.6.7 logs an error when using
(built-in) bindings with unsupported output modes.

The primary cause for this is that the output modes are restricted from
outputting these events to the operating system.

See [OpenTabletDriver issue #2078](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/2078)
for more details.

> ### Multikey bindings doesn't respect keyboard layout {#multikey-binding-keyboard-layout}

Reviewer: ```suggestion
### Multikey bindings don't respect keyboard layout {#multikey-binding-keyboard-layout}
```

For non-QWERTY users (e.g. AZERTY), you may experience that keyboard bindings

> are read correctly in the GUI, but sent as a QWERTY keycode by the daemon.

Reviewer: Probably worth elaborating a bit more on this.

```suggestion
For non-QWERTY users (e.g. AZERTY or QWERTZ), you may experience that keyboard bindings
are read correctly in the GUI, but sent as a QWERTY keycode by the daemon.

This occurs because Eto reads filtered keycodes for setting bindings in the GUI but OpenTabletDriver outputs the raw keycodes associated with these bindings. The operating system then applies your keyboard/language filtering to the raw keycode, transforming it into something unexpected.
```

layout.

In that case, follow the workaround presented for
[System-wide keyboard layout](#multikey-binding-keyboard-layout-system-wide-layouts) users.

If it turns out that this doesn't work anyway, make sure the virtual keyboard
from OpenTabletDriver is assigned the US-American layout.

##### Sway window manager {#multikey-binding-keyboard-layout-linux-sway}

Put this *after* any other keyboard configuration:

```conf
input '0:0:OpenTabletDriver_Virtual_Keyboard' {
    xkb_layout us
}
```

### Some tablets may not work coming out of system sleep/suspend {#sleep-wakeup-is-inconsistent}

Replugging the device always fixes this.

See [OpenTabletDriver issue #3171](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/3171)
for more details.

### Disconnecting tablets mid-event (e.g. button press) causes the event to remain held (button permanently pressed) {#disconnected-tablets-do-not-reset-pipeline}

The only known workaround is to restart the daemon or send the same type of
event again (e.g. from the reconnected or another device)

See [OpenTabletDriver issue #2955](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/2957)

> for more details.

Reviewer: I'm not sure this is a thing anymore. Needs to be checked again.

Author: It does still exist, but the link has a mistake in the title so thanks for highlighting this regardless.

Author: You can test with holding down your pen somewhere, disconnecting your tablet, and then moving your mouse around. At least on Sway this causes my left click to remain stuck in Firefox.

See the upstream issue here: https://github.com/OpenTabletDriver/OpenTabletDriver/issues/2957

### (Linux) Flatpak release does not install udev rules or kernel module blacklist {#flatpak-missing-permissions}

This means that the Flatpak release may show no tablets detected, caused by
missing permissions or conflicting drivers.

Follow the instructions from the description of the Flatpak package to set these up properly.

See [OpenTabletDriver issue #3266](https://github.com/OpenTabletDriver/OpenTabletDriver/issues/3266)

> for more details.

Reviewer: Maybe should note that this cannot be set up automatically due to how flatpak works. Not really a bug.

Author: I still think this is something we can better explain or do inside of the driver. There's a few things I imagine we could do but not limited to:

- Probing for known udev rule locations when known tablets exists but aren't accessible (present in HID list but `CanOpen: false` or similar)
- Requesting exclusive access to the device via `polkit` (or a setuid binary)
- Disabling the device in libinput manually (especially for `LIBINPUT_IGNORE_DEVICE`)

Regardless we can definitely put this tidbit elsewhere as issue #322 already exists to track the lack of this documentation.

Fixed in 25a47c7d5

## Assorted OpenTabletDriver Bugs {#general-bugs}

> These bugs are unlikely to be encountered by users, but have been seen in the wild

Reviewer: ```suggestion
These bugs are unlikely to be encountered by users, but have been seen in the wild.
```
